datasets/swda/swda.py

```diff
@@ -169,35 +427,336 @@ def _split_generators(self, dl_manager):
+        # Read in corpus data using split files.
+        corpus = CorpusReader(src_dirname=data_dir, split_file=split_file)
+        # Generate examples.
+        for i_trans, trans in enumerate(corpus.iter_transcripts()):
+            for i_utt, utt in enumerate(trans.utterances):
+                id_ = i_trans + i_utt
+                yield id_, {
+                    "swda_filename": utt.swda_filename,
+                    "conversation_no": utt.conversation_no,
+                    "utterance_index": utt.utterance_index,
+                    "subutterance_index": utt.subutterance_index,
+                    "transcript_index": utt.transcript_index,
+                    "act_tag": utt.act_tag,
+                    "damsl_act_tag": utt.damsl_act_tag(),
+                    "text": utt.text,
+                    "pos": utt.pos,
+                    "topic_description": utt.topic_description,
+                    "caller": utt.caller,
+                    "trees": utt.trees,
+                    "ptb_basename": utt.ptb_basename,
+                    "ptb_treenumbers": utt.ptb_treenumbers,
+                }
 
-                # Parse row into speaker info | utterance text | dialogue act tag.
-                parsed_row = row.rstrip("\r\n").split("|")
 
-                # Also returning dialogue act integer label.
-                yield f"{split}-{id_}", {
-                    "speaker": parsed_row[0],
-                    "utterance_text": parsed_row[1],
-                    "dialogue_act_tag": parsed_row[2],
-                }
+class CorpusReader:
+    """Class for reading in the corpus and iterating through its values."""
+
+    def __init__(self, src_dirname, split_file=None):
+        """
+        Reads in the data from `src_dirname` (should be the root of the
+        corpus).  Assumes that the metadata file `swda-metadata.csv` is
+        in the main directory of the corpus, using that file to build
+        the `Metadata` object used throughout.
+
+        Added split_files to separate between splits. If given None it will use all data files.
+        """
+        self.src_dirname = src_dirname
+        metadata_filename = os.path.join(src_dirname, "swda-metadata.csv")
+        self.metadata = Metadata(metadata_filename)
+        self.split_file = split_file
+
+    def iter_transcripts(
+        self,
+    ):
+        """
+        Iterate through the transcripts.
+        """
+
+        # All files names.
+        filenames = glob.glob(os.path.join(self.src_dirname, "sw*", "*.csv"))
+        # If no split files are mentioned just use all files.
+        self.split_file = filenames if self.split_file is None else self.split_file
+        # Filter out desired file names
+        filenames = [
+            file for file in filenames if os.path.basename(file).split("_")[-1].split(".")[0] in self.split_file
+        ]
+        for filename in filenames:
```

Comment:
```suggestion
        for filename in sorted(filenames):
```
we need to make the order deterministic here (glob may not always return the files in the same order)